app/services/katello/pulp/repository.rb

```diff
@@ -0,0 +1,23 @@
+module Katello
+  module Pulp
+    class Repository < ::Actions::Pulp::Abstract
+      attr_accessor :repo, :input
+
+      def initialize(repo, input)
+        @repo = repo
+        @input = input
+      end
+
+      def sync
+        sync_options = {}
+        sync_options[:max_speed] ||= SETTINGS[:katello][:pulp][:sync_KBlimit] if SETTINGS[:katello][:pulp][:sync_KBlimit]
+        sync_options[:num_threads] ||= SETTINGS[:katello][:pulp][:sync_threads] if SETTINGS[:katello][:pulp][:sync_threads]
+        sync_options[:feed] = @input[:source_url] if @input[:source_url]
+        sync_options[:validate] = !(SETTINGS[:katello][:pulp][:skip_checksum_validation])
```

Comment:
I would use the dig method on these SETTINGS https://ruby-doc.org/core-2.3.0_preview1/Hash.html#method-i-dig